src/test/unit_tests/test_bucketspace.js

```diff
@@ -41,6 +41,169 @@ const DEFAULT_FS_CONFIG = {
+            await rpc_client.bucket.delete_bucket({ name: bucket });
+            await rpc_client.account.delete_account({ email: `${account_name}@noobaa.com` });
+        }
+        await fs_utils.folder_delete(tmp_fs_root);
+    });
+
+    mocha.it('list buckets - each account can list only its own bucket - no bucket policies applied ', async function() {
+        for (const account of Object.values(accounts)) {
+            const { s3_client } = account;
+            const res = await s3_client.listBuckets({});
+            const buckets = res.Buckets.filter(bucket => bucket.Name !== 'first.bucket');
+            assert.equal(buckets.length, 1);
+            assert.equal(buckets[0].Name, account.bucket);
+        }
+    });
+    mocha.it('account1 - set allow all policy, all account can list bucket1', async function() {
+        // Give account access to the required buckets
+        const public_bucket = accounts.account1.bucket;
+        await Promise.all(
+            [public_bucket]
+            .map(bucket => test_utils.generate_s3_policy('*', bucket, ['s3:*']))
+            .map(generated =>
+                rpc_client.bucket.put_bucket_policy({
+                    name: generated.params.bucket,
+                    policy: generated.policy,
+                })
+            )
+        );
+        for (const account of Object.values(accounts).slice(1)) {
+            const { s3_client } = account;
+            const res = await s3_client.listBuckets({});
+            const buckets = res.Buckets.filter(bucket => bucket.Name !== 'first.bucket' && bucket.Name !== account.bucket);
+            assert.equal(buckets.length, 1);
+            assert.equal(buckets[0].Name, public_bucket);
+        }
+
+        // delete bucket policy
+        await rpc_client.bucket.put_bucket_policy({
+            name: public_bucket,
+            policy: '',
+        });
+    });
+
+    mocha.it('account2 - set allow all only account1, account1 should list bucket2 but account3 shouldnt', async function() {
+        // Give account access to the required buckets
+        const bucket2 = accounts.account2.bucket;
+        const account_name = 'account1';
+        // on NC the account identifier is account name, and on containerized it's the account's email
+        const principal = process.env.NC_CORETEST ? account_name : `${account_name}@noobaa.com`;
+        await Promise.all(
+            [bucket2]
+            .map(bucket => test_utils.generate_s3_policy(principal, bucket, ['s3:*']))
+            .map(generated =>
+                rpc_client.bucket.put_bucket_policy({
+                    name: generated.params.bucket,
+                    policy: generated.policy,
+                })
+            )
+        );
+        let s3_client = accounts.account1.s3_client;
+        let res = await s3_client.listBuckets({});
+        let buckets = res.Buckets.filter(bucket => bucket.Name !== 'first.bucket' && bucket.Name !== accounts.account1.bucket);
```

Comment:
Why did you add the second condition `bucket.Name !== accounts.account1.bucket`?
Please add an explanation in a comment

Reply:
I removed it